htdocs/lib/MapMarkers.js

```diff
@@ -0,0 +1,680 @@
+        );
+    }
+};
+
+MarkerManager.prototype.find = function(id) {
+    return id in this.markers? this.markers[id] : null;
+};
+
+MarkerManager.prototype.add = function(id, marker) {
+    this.markers[id] = marker;
+};
+
+MarkerManager.prototype.ageAll = function() {
+    var now = new Date().getTime();
+    var data = this.markers;
+    $.each(data, function(id, x) {
+        if (!x.age(now - x.lastseen)) delete data[id];
+    });
+};
+
+MarkerManager.prototype.clear = function() {
+    // Remove all markers from the map
+    $.each(this.markers, function(_, x) { x.setMap(); });
+    // Delete all markers
+    this.markers = {};
+};
+
+//
+// Generic Map Marker
+// Derived classes have to implement:
+//     setMap(), setMarkerOpacity()
+//
+
+function Marker() {}
+
+// Wrap given callsign or other ID into a clickable link.
+Marker.linkify = function(name, url = null, linkEntity = null) {
+    // If no specific link entity, use the ID itself
+    if (linkEntity == null) linkEntity = name;
+
+    // Must have valid ID and lookup URL
+    if ((name == '') || (url == null) || (url == '')) {
+        return name;
+    } else {
+        return '<a target="callsign_info" href="' +
+            url.replaceAll('{}', linkEntity) + '">' + name + '</a>';
+    }
+};
+
+// Create link to tune OWRX to the given frequency and modulation.
+Marker.linkifyFreq = function(freq, mod) {
+    var text;
+    if (freq >= 30000000) {
+        text = "" + (freq / 1000000.0) + "MHz";
+    } else if (freq >= 10000) {
+        text = "" + (freq / 1000.0) + "kHz";
+    } else {
+        text = "" + freq + "Hz";
+    }
+
+    return '<a target="openwebrx-rx" href="/#freq=' + freq +
+        ',mod=' + mod + '">' + text + '</a>';
+};
+
+// Compute distance, in kilometers, between two latlons.
+Marker.distanceKm = function(p1, p2) {
+    // Earth radius in km
+    var R = 6371.0;
+    // Convert degrees to radians
+    var rlat1 = p1.lat() * (Math.PI/180);
+    var rlat2 = p2.lat() * (Math.PI/180);
+    // Compute difference in radians
+    var difflat = rlat2-rlat1;
+    var difflon = (p2.lng()-p1.lng()) * (Math.PI/180);
+    // Compute distance
+    d = 2 * R * Math.asin(Math.sqrt(
+        Math.sin(difflat/2) * Math.sin(difflat/2) +
+        Math.cos(rlat1) * Math.cos(rlat2) * Math.sin(difflon/2) * Math.sin(difflon/2)
+    ));
+    return Math.round(d);
+};
+
+// Truncate string to a given number of characters, adding "..." to the end.
+Marker.truncate = function(str, count) {
+    return str.length > count? str.slice(0, count) + '&mldr;' : str;
+};
+
+// Convert degrees to compass direction.
+Marker.degToCompass = function(deg) {
+    dir = ['N', 'NNE', 'NE', 'ENE', 'E', 'ESE', 'SE', 'SSE', 'S', 'SSW', 'SW', 'WSW', 'W', 'WNW', 'NW', 'NNW'];
+    return dir[Math.floor((deg/22.5) + 0.5) % 16];
+};
+
+// Convert given name to an information section title.
+Marker.makeListTitle = function(name) {
+    return '<div style="border-bottom:2px solid;"><b>' + name + '</b></div>';
```

Comment:
in general, i'd prefer CSS for styling (instead of html `<b>` markup).

Reply:
That would prevent someone from mixing bold and normal inside that div in the future.

Reply:
Would that make it better or worse?

If you need the flexibility of being able to switch between bold or not, I would suggest adding a class to distinguish between the cases.

I'm not really at a point where I do understand every bit of this yet, so this is more of a general, best-practice suggestion.

Reply:
Essentially, I use <b> / <i> / etc inside the text if I am simply planning to emphasize something. Using a style or a CSS class in such cases feels unnecessarily cumbersome in such cases. I can set bold style on the whole <div> of course, but this means the whole title will always remain bold.